### ISSUE TITLE
Unexpected error during shrinking with nested `combine` arbitraries

### ISSUE THREAD

**Issue by szktty** [szktty/kiri-check#23]
An unexpected error occurs during shrinking when using nested `combine` arbitraries. This issue seems to be related to the return value of the `transformer` function passed to `combine`.

## Steps to reproduce

The issue can be reproduced with the following code:

```dart
property('good', () {
  forAll(
    combine2(
      combine2(integer(), integer(), (a, b) => a + b),
      integer(),
      (a, b) => a + b,
    ),
    (value) {
      fail('bug');
    },
  );
});

property('bad', () {
  forAll(
    combine2(
      combine2(integer(), integer(), (a, b) => a * b),
      integer(),
      (a, b) => a + b,
    ),
    (value) {
      fail('bug');
    },
  );
});

property('bad', () {
  forAll(
    combine2(
      combine2(integer(), integer(), (a, b) => a + b),
      integer(),
      (a, b) => a * b,
    ),
    (value) {
      fail('bug');
    },
  );
});
```

## Error message

```
dart:core                                                        List.[]
package:kiri_check/src/arbitrary/combinator/combine.dart 190:65  CombineArbitrary.shrink
package:kiri_check/src/arbitrary/combinator/combine.dart 193:42  CombineArbitrary.shrink
package:kiri_check/src/property.dart 327:36                      ShrinkingContext.shrink
package:kiri_check/src/property.dart 104:29                      StatelessProperty.check
===== asynchronous gap ===========================
package:kiri_check/src/property.dart 486:9                       PropertyTestRunner.run
===== asynchronous gap ===========================
package:kiri_check/src/property.dart 466:5                       PropertyTestManager.runTest
===== asynchronous gap ===========================
package:kiri_check/src/property.dart 392:7                       PropertyTest.register.body
RangeError (index): Invalid value: Valid value range is empty: 0
```
